pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -322,35 +307,45 @@ func nsgRulesAreValid(existingRules []network.SecurityRule, desiredRules []netwo
 	return true
 }
 
-func patchNsgRules(existingRules []network.SecurityRule, desiredRules []network.SecurityRule) []network.SecurityRule {
-	result := make([]network.SecurityRule, 0, len(existingRules))
+func addOrReplaceNsgRulesDefinition(existingRules *[]network.SecurityRule, desiredRules *[]network.SecurityRule) {
+	if existingRules == nil || desiredRules == nil {
+		return
+	}
```

Comment:
to check existing security group Rules whether are empty in a security group.  in case a security group is created but security group rules are not created or a part of creating.  the empty security group rules are not treated as an error here.  it will move to next to create security group rules if empty or patch the existing rules